src/common/downloadFileDialog.component.tsx

```diff
@@ -0,0 +1,120 @@
+import WarningIcon from '@mui/icons-material/Warning';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormHelperText,
+} from '@mui/material';
+import { UseQueryResult } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import React from 'react';
+import { APIImage, APIImageWithURL } from '../api/api.types';
+import { downloadFileByLink } from '../utils';
+
+export interface BaseDownloadFileProps {
+  open: boolean;
+  onClose: () => void;
+  fileType: 'Image' | 'Attachment';
+}
+
+export interface ImageDownloadDialogProps extends BaseDownloadFileProps {
+  fileType: 'Image';
+  useGetFile: (id: string) => UseQueryResult<APIImageWithURL, AxiosError>;
+  file: APIImage;
+}
+
+export type DownloadFileProps = ImageDownloadDialogProps;
+
+const DownloadFileDialog = (props: DownloadFileProps) => {
+  const { open, onClose, fileType, file, useGetFile } = props;
+
+  const [formError, setFormError] = React.useState<string | undefined>(
+    undefined
+  );
+
+  const [downloadId, setDownloadId] = React.useState<string>('');
+
+  const {
+    data: downloadFile,
+    isLoading: downloadIsLoading,
+    error,
+    isPending: downloadIsPending,
+  } = useGetFile(downloadId);
+
+  const handleClose = React.useCallback(() => {
+    setFormError(undefined);
+    setDownloadId('');
+    onClose();
+  }, [onClose]);
+
+  const handleClick = React.useCallback(() => {
+    setDownloadId(file.id);
+  }, [file]);
+
+  const handleDownloadFile = React.useCallback(() => {
+    if (!error && downloadFile) {
+      downloadFileByLink(
+        document,
+        downloadFile.download_url,
+        downloadFile.file_name
+      );
+      setDownloadId('');
+      onClose();
+    } else {
+      setFormError('No data provided. Please refresh and try again');
+    }
+  }, [downloadFile, onClose, error]);
+
+  React.useEffect(() => {
+    if (!downloadIsPending) {
+      handleDownloadFile();
+    }
+  }, [downloadFile, error, handleDownloadFile, downloadIsPending]);
```

Comment:
Instead of fetching data through the `useEffect`, use `QueryClient.fetchQuery` (https://tanstack.com/query/latest/docs/reference/QueryClient/#queryclientfetchquery). This approach minimizes the risk of errors related to the dependencies in the `useEffect` hook  and ensures more predictable behavior. This would probably require a redesign to find the most optimal solution for the this use case.